portfolio/src/main/java/com/google/sps/data/BookDaoDatastore.java

```diff
@@ -0,0 +1,206 @@
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+/** 
+* Access Datastore to manage Books
+*/
+public class BookDaoDatastore implements BookDao {
+
+    private static final String ENTITY_KIND = "Book";
+    private static final String TIME_STAMP = "timeStamp";
+    private static final String TITLE = "title";
+    private static final String GENRE = "genre";
+    private static final String CATEGORIES = "categories";
+    private static final String AUTHORS = "authors";
+    private static final String LANGUAGE = "language";
+    private static final String DESCRIPTION = "description";
+    private static final String INFO_LINK = "infoLink";
+    private static final String THUMBNAIL = "thumbnail";
+    private static final String PAGE_COUNT = "pageCount";
+    private static final String PUBLISHED_DATE = "publishedDate";
+    private static final String PUBLISHER = "publisher";
+    private static final String MATURITY_RATING = "maturityRating";
+    private static final String ISBN = "isbn";
+
+    private DatastoreService datastore;
+
+    public BookDaoDatastore() {
+        datastore = DatastoreServiceFactory.getDatastoreService();
+    }
+
+    /**
+    * Create a Book entity in Datastore
+    * @param book: the book that you want to create an entity from
+    */
+    @Override
+    public void create(Book book){
+      if(validate(book)){
+        if(!entityExists(book.isbn())){
+          datastore.put(parseEntity(book));
+        }else{
+          throw new RuntimeException(String.format("Book already exists with ISBN-13=%s", book.isbn()));
+        }
+      }
+    }
+
+    /**
+    * Delete a Book entity in Datastore
+    * @param isbn: the isbn of the book that you want to delete
+    */
+    @Override
+    public void delete(String isbn){
+        if(validate(isbn) && entityExists(isbn)){
+          datastore.delete(createKey(isbn));
+        }
+    }
+
+    /**
+    * Gets the entity from datastore and parses as Book
+    * @param isbn: The isbn of the book you want
+    */
+    @Override
+    public Book getEntity(String isbn) {
+        if(validate(isbn)){
+            Entity bookEntity;
+            try{
+                bookEntity = datastore.get(createKey(isbn));
+            } catch(EntityNotFoundException ex) {
+                return null;
+            }
+            return parseBook(bookEntity);
+        }
+        return null;
+    }
+
+    /**
+    * Update a user entity in Datastore
+    * @param book: The book to be updated
+    */
+    @Override
+    public void update(Book book) {
+        if(validate(book) && entityExists(book.isbn())){
+            datastore.put(parseEntity(book));
+        }
```

Comment:
validate always return true, so it is not useful to check the return value.  Also, you should throw an exception if the update fails.
```
validate(book);
if(entityExists(book.isbn())){
  datastore.put(parseEntity(book));
} else {
  throw new RuntimeException(String.format("Update book: Book not found with isbn=%s", book.isbn);
}
``